src/compiler/compile/render_ssr/handlers/DynamicElement.ts

```diff
@@ -0,0 +1,215 @@
+import {
+	get_attribute_value,
+	get_class_attribute_value
+} from './shared/get_attribute_value';
+import { get_slot_scope } from './shared/get_slot_scope';
+import { boolean_attributes } from './shared/boolean_attributes';
+import Renderer, { RenderOptions } from '../Renderer';
+import DynamicElement from '../../nodes/DynamicElement';
+import ElementHandler from './Element';
+import { x } from 'code-red';
+import Expression from '../../nodes/shared/Expression';
+import remove_whitespace_children from './utils/remove_whitespace_children';
+import Element from '../../nodes/Element';
+import { Expression as ESExpression } from 'estree';
+
+export default function (
+	node: DynamicElement,
+	renderer: Renderer,
+	options: RenderOptions & {
+		slot_scopes: Map<any, any>;
+	}
+) {
+	const dependencies = node.tag.dynamic_dependencies();
+
+	if (dependencies.length === 0) {
+		((node as unknown) as Element).dynamic_tag = node.tag;
```

Comment:
Can `DynamicElement` extend `Element` or perhaps there should be a common ancestor like `BaseElement` created?